### PR TITLE
Added Tests for Object Detection C APIs

### PR DESCRIPTION
**Pull request by priankakariatyml**

### CODE REVIEW

File: tensorflow_lite_support/c/test/task/vision/object_detector_test.cc [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+  EXPECT_NE(detection_result->detections, nullptr);
+  EXPECT_GE(detection_result->detections->size, 1);
+  EXPECT_NE(detection_result->detections[0].categories, nullptr);
+  EXPECT_EQ(strcmp(detection_result->detections[0].categories[0].label,
+                   allowlisted_label_name),
+            0);
+
+  TfLiteDetectionResultDelete(detection_result);
+}
+
```
Comment (khanhlvg) [inline tensorflow_lite_support/c/test/task/vision/object_detector_test.cc:382]: Can you add test for score threshold?

**Review by khanhlvg — CHANGES_REQUESTED** [review]

File: tensorflow_lite_support/c/test/task/vision/object_detector_test.cc [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+constexpr char kMobileSsdWithMetadata[] =
+    "coco_ssd_mobilenet_v1_1.0_quant_2018_06_29.tflite";
+
+StatusOr<ImageData> LoadImage(const char* image_name) {
+  return DecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name));
+}
+
+void VerifyDetection(TfLiteDetection& detection, TfLiteBoundingBox bounding_box,
+                     float first_category_score, char* first_category_label) {
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/test/task/vision/object_detector_test.cc:52]: nit: bounding_box -> expected_bounding_box, first_category_score -> expected_first_score, first_category_label -> expected_first_label.

File: tensorflow_lite_support/c/test/task/vision/object_detector_test.cc [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+  EXPECT_GE(detection.size, 1);
+  EXPECT_NE(detection.categories, nullptr);
+  EXPECT_EQ(detection.bounding_box.origin_x, bounding_box.origin_x);
+  EXPECT_EQ(detection.bounding_box.origin_y, bounding_box.origin_y);
+  EXPECT_EQ(detection.bounding_box.height, bounding_box.height);
+  EXPECT_EQ(detection.bounding_box.width, bounding_box.width);
+
+  EXPECT_EQ(strcmp(detection.categories[0].label, first_category_label), 0);
```
Comment (lu-wang-g) [inline tensorflow_lite_support/c/test/task/vision/object_detector_test.cc:60]: EXPECT_THAT(detection.categories[0].label, StrEq(first_category_label));

File: tensorflow_lite_support/c/test/task/vision/object_detector_test.cc (hunk 1 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/c/test/task/vision/object_detector_test.cc:384]: Donee

File: tensorflow_lite_support/c/test/task/vision/object_detector_test.cc (hunk 3 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/c/test/task/vision/object_detector_test.cc]: Fixed

**Review by khanhlvg — APPROVED** [review]

**Pull request merged**